### PR TITLE
 Don't output EXPOSE if empty port specified

### LINKED ISSUE #391: Is Empty PORT EXPOSE By Design
I was generating a Dockerfile with the `> Dockerfile` command, and decided to leave the `PORT` section as an empty string, as I'm generating a base image that doesn't require exposing. This is what is generated 
screen shot 2018-08-22 at 14 37 00

I know it's as easy as just deleting the last line, but I'm wondering if this is by design, or if someone doesn't specify a port, EXPOSE should not be generated.

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Fixes #391 

For next milestone.

### CODE REVIEW

File: configureWorkspace/configure.ts [hunk 1]
```diff
@@ -53,11 +53,18 @@ export type ConfigureTelemetryProperties = {
...
+export interface IPlatformGeneratorInfo {
     genDockerFile: GeneratorFunction,
     genDockerCompose: GeneratorFunction,
-    genDockerComposeDebug: GeneratorFunction
-}>();
+    genDockerComposeDebug: GeneratorFunction,
+    defaultPort: string
+}
+
+export function getExposeStatements(port: string): string {
```
Comment (PrashanthCorp) [inline configureWorkspace/configure.ts]: getExposeStatement? I don't see the reasoning behind the plural name.

Comment (StephenWeatherford) [reply configureWorkspace/configure.ts]: VS has scenarios where there are more than one (HTTPS and HTTP ports).  I'm trying to keep similar to their logic.

File: configureWorkspace/configure_go.ts [hunk 2]
```diff
@@ -3,15 +3,18 @@
...
+export let configureGo: IPlatformGeneratorInfo = {
   genDockerFile,
   genDockerCompose,
-  genDockerComposeDebug
+  genDockerComposeDebug,
+  defaultPort: '3000'
 };
 
 function genDockerFile(serviceNameAndRelativePath: string, platform: string, os: string | undefined, port: string, { cmd, author, version, artifactName }: Partial<PackageInfo>): string {
+  let exposeStatements = getExposeStatements(port);
```
Comment (EricJizbaMSFT) [inline configureWorkspace/configure_go.ts]: There seems to be an inconsistency where sometimes the `genDockerFile` has a 'port' parameter and sometimes it has an 'exposeStatements' parameter. Can we make that consistent?

Comment (StephenWeatherford) [reply configureWorkspace/configure_go.ts]: done

File: configureWorkspace/configure.ts [hunk 3]
```diff
@@ -70,7 +77,14 @@ function genDockerFile(serviceNameAndRelativePath: string, platform: Platform, o
     let generators = generatorsByPlatform.get(platform);
     assert(generators, `Could not find dockerfile generator functions for "${platform}"`);
     if (generators.genDockerFile) {
-        return generators.genDockerFile(serviceNameAndRelativePath, platform, os, port, { cmd, author, version, artifactName });
+        let contents = generators.genDockerFile(serviceNameAndRelativePath, platform, os, exposeStatements, { cmd, author, version, artifactName });
```
Comment (EricJizbaMSFT) [inline configureWorkspace/configure.ts]: Looks like you missed changing this 'exposeStatements' back to 'port'

**Pull request merged**